### ISSUE TITLE
CMake 3.16.4 & Windows: Manually-specified variables were not used by the project

### ISSUE THREAD

**Issue by jplatte** [alexcrichton/cmake-rs#92]
I currently can't compile a C++ project with cmake-rs on Windows because it sets `CMAKE_BUILD_TYPE`, which is not used when CMake generates build files for Visual Studio. This is the error message I get:

```
error: failed to run custom build command for `cratename v0.1.0 (C:\crate\path)`

Caused by:
  process didn't exit successfully: 'C:\crate\path\debug\build\cratename-hash\build-script-build' (exit code: 101)
--- stderr
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_BUILD_TYPE

thread 'main' panicked at '
command did not execute successfully, got: exit code: 1

build script failed, must exit now', C:\Users\Jonas Platte\.cargo\registry\src\github.com-1eccc6299db9ec823\cmake-0.1.42\src\lib.rs:861:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I am not setting `CMAKE_BUILD_TYPE` manually. Happens in debug and release.

**Comment by jplatte** [comment]
Actually, that warning doesn't seem to be the cause for the non-zero exit code. Adding a dummy use of `CMAKE_BUILD_TYPE` to my `CMakeLists.txt`* gets rid of the warning, but the build still fails, now without any messages from CMake...

\* [inspired by StackOverflow](https://stackoverflow.com/questions/36451368/get-rid-of-cmake-warning-manually-specified-variables-were-not-used-by-the-proj)

**Comment by raf924** [comment]
@jplatte did you eventually manage to have a successful build with cmake-rs ? if so, how ?

**Comment by jplatte** [comment]
@raf924 I did. I can't remember what exactly the issue was, but all I had to do was just reproduce it outside of cmake-rs by passing the same flags and then I actually got some useful error messages. It was an error in my `CMakeLists.txt` in the end.

I still kept the dummy use of `CMAKE_BUILD_TYPE`, though I'm not certain whether it's actually needed.

**Comment by raf924** [comment]
Welp, I just tried that. Copied and adapted the cmakecache.txt generated by cmake-rs and ran cmake on it. Builds flawlessly but with cmake-rs it fails for seemingly no reason. Checked the Cmakeoutput.log of both cmake-rs and cmake and it's basically the same with no error or even a warning. It's too bad. I can use the cc crate instead this time but I would have liked to import a cmake-based project into my program easily.
